uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRendererHelper.java

```diff
@@ -138,9 +158,18 @@ public double getWidth(final List<GridColumn<?>> columns) {
     public RenderingInformation getRenderingInformation() {
         final GridData model = view.getModel();
         final Bounds bounds = getVisibleBounds();
-        final List<GridColumn<?>> allColumns = new ArrayList<GridColumn<?>>();
-        final List<GridColumn<?>> bodyColumns = new ArrayList<GridColumn<?>>();
-        final List<GridColumn<?>> floatingColumns = new ArrayList<GridColumn<?>>();
+        final List<GridColumn<?>> allColumns = new ArrayList<>();
+        final List<GridColumn<?>> bodyColumns = new ArrayList<>();
+        final List<GridColumn<?>> floatingColumns = new ArrayList<>();
+
+        double viewHeight = 0;
+        final int rowCount = model.getRowCount();
+        final List<Double> rowHeights = new ArrayList<>();
+        for (GridRow row : model.getRows()) {
+            final double rowHeight = row.getHeight();
```

Comment:
Is this `getHeight()` in the same slowpath?

Reply:
No really. This use is in the `getRenderingInformation()` method that gets the _all row heights_ collection used elsewhere (to avoid this same loop being called again and again elsewhere)... So, yes, it is the same loop as has been replaced but now it should be the only loop of this nature!